hack/generate-crds.sh

```diff
@@ -148,14 +147,6 @@ generate_group () {
     mv "$crd" "$crd_out"
     relevant_files+=("$(basename "$crd_out")")
 
-    if $add_deletion_protection_label; then
-      if grep -q "clusters.extensions.gardener.cloud"  "$crd_out"; then
-        :
-      else
-        sed -i '4 a\  labels:\n\    gardener.cloud/deletion-protected: "true"' "$crd_out"
```

Comment:
What if extensions are not using crds deployer to deploy CRDs in that case we should not remove it from here.